### ISSUE TITLE
[BUG] Only one subtag per file is represented as deck

### ISSUE THREAD

**Issue by phylome** [st3v3nmw/obsidian-spaced-repetition#691]
**Describe the bug**
When defining subtags in a file to represent flashcards in different decks, only one, namely the first in the file, is represented as a card deck. The others don't appear.

**To Reproduce**

1. Create a file `Unit 5` with the following content:

```
# Vocab - the football match

#English/Unit-5/Vocab

gewinnen:::to win
verlieren:::to lose

# Grammar - modal verbs

#English/Unit-5/Grammar

must::to have to
may::to be allowed to
```

2. Issue command `Select a deck to cram` or `Review flashcards`
3. The following deck hierarchy appears:
- English
    - Unit-5
        - Vocab

_Grammar_ is missing.

**Expected behavior**
Two decks will appear with the following hierarchy
- English
    - Unit-5
        - Grammar
        - Vocab

**Versions (please complete the following information):**

-   OS: Linux Ubuntu 20.04
-   Obsidian version: 1.3.4
-   Plugin version: 1.10.1

**Additional context**
Add any other context about the problem here (e.g. the markdown producing the error).

**Comment by sam-messina** [comment]
is there any workaround for this? seems like there should be a simple way to separate with an <end-tag> line, so you can keep multiple decks in the same file

**Comment by Lucas256-cmd** [comment]
I have the same problem. I thought a tag was valid until the next tag or next heading but it is valid for the whole file I think.

**Comment by ronzulu** [comment]
Hi all, I have completed the update but would appreciate beta testing and feedback before I finalise the PR.

The draft PR together with the updated code is available at:
https://github.com/st3v3nmw/obsidian-spaced-repetition/pull/834

Cheers
Ronny

**Comment by ronzulu** [comment]
Released into 1.12.0
